### PR TITLE
drop guava dependency to 11.0.2

### PR DESCRIPTION
**Pull request by danielnorberg**
Simplify guava dependency conflict resolution in applications using this library.

### CODE REVIEW

File: src/main/java/com/spotify/dns/RetainingDnsSrvResolver.java [hunk 1]
```diff
@@ -45,14 +43,16 @@
...
     try {
-      List<LookupResult> nodes = delegate.resolve(fqdn);
+      final List<LookupResult> nodes = delegate.resolve(fqdn);
 
+      // No nodes resolved? Return stale data.
       if (nodes.isEmpty()) {
-        nodes = firstNonNull(cache.get(fqdn), nodes);
-      } else {
-        cache.put(fqdn, nodes);
+        List<LookupResult> first = cache.get(fqdn);
```
Comment (pettermahlen) [inline src/main/java/com/spotify/dns/RetainingDnsSrvResolver.java]: Rename 'first' to 'cached'?

**Comment by pettermahlen** [comment]
I guess the rationale for going as old as 11.0.2 is that you expect fewer runtime conflicts? I think it makes sense to document the reasoning for why you expect that in a comment in this PR.

**Comment by protocol7** [comment]
+1

File: src/main/java/com/spotify/dns/LookupResult.java [hunk 2]
```diff
@@ -1,19 +1,96 @@
...
+    if (priority != that.priority) {
+      return false;
+    }
+    if (weight != that.weight) {
+      return false;
+    }
+    if (ttl != that.ttl) {
+      return false;
+    }
+    return !(host != null ? !host.equals(that.host) : that.host != null);
```
Comment (marcusb) [inline src/main/java/com/spotify/dns/LookupResult.java]: Objects.equal?

Comment (danielnorberg) [inline src/main/java/com/spotify/dns/LookupResult.java]: Intellij generated code. I'm lazy but can change it if you feel strongly about it.

File: src/main/java/com/spotify/dns/RetainingDnsSrvResolver.java (hunk 1 above)
Comment (danielnorberg) [inline src/main/java/com/spotify/dns/RetainingDnsSrvResolver.java]: Done.

**Comment by marcusb** [comment]
+1

@pettermahlen: For me, AutoValue was a bit overkill for this simple class, considering the Eclipse integration is so painful. Also it would make sense to add a toHostAndPort() getter which is prevented by AutoValue.

**Comment by pettermahlen** [comment]
@marcusb i don't think it's correct that AutoValue prevents adding a toHostAndPort() method? 

```
public HostAndPort toHostAndPort() {
   return HostAndPort.fromParts(host(), port());
}
```

But if Eclipse doesn't support annotation processors well, then that's perhaps an issue - it's a non-issue with IntelliJ, it just works. 

@danielnorberg do you feel that you know that older is better for Guava?

**Comment by danielnorberg** [comment]
Yeah, older guava makes satisfying the dependency easier for users of this library.

**Comment by pettermahlen** [comment]
To clarify what I mean: I think that reducing the number of 'mutating methods in Guava' used by this library minimises the risk for users of dns-java. But I'm not sure that it's better to use Guava 11 than 15? What makes you say that that's better? AFAICT, compatibility issues go both forwards and backwards in time, so maybe in some way, the version of Guava that is most commonly used might be the one that is best from a compatibility perspective? 

Perhaps I'm missing something. Probably, it's not possible to objectively find a 'best' Guava version to depend on.

**Comment by danielnorberg** [comment]
Not saying that some version of guava is inherently better. Just that for some users of this library it'll be easier to satisfy e.g. an upper bounds dependency constraint (https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html).

**Comment by pettermahlen** [comment]
Right, that makes sense.

**Pull request merged**